src/psyclone/tests/psyir/symbols/symboltable_test.py

```diff
@@ -136,6 +137,79 @@ def test_add():
             "'var1'.") in str(error.value)
 
 
+def test_imported_symbols():
+    ''' Test the imported_symbols method. '''
+    sym_table = SymbolTable()
+    my_mod = ContainerSymbol("my_mod")
+    sym_table.add(my_mod)
+    assert sym_table.imported_symbols(my_mod) == []
+    var1 = DataSymbol("var1", DataType.REAL, interface=LocalInterface())
+    sym_table.add(var1)
+    assert sym_table.imported_symbols(my_mod) == []
+    var2 = DataSymbol("var2", DataType.INTEGER,
+                      interface=GlobalInterface(my_mod))
+    assert sym_table.imported_symbols(my_mod) == []
+    sym_table.add(var2)
+    assert sym_table.imported_symbols(my_mod) == [var2]
+    var3 = DataSymbol("var3", DataType.INTEGER,
+                      interface=GlobalInterface(my_mod))
+    sym_table.add(var3)
+    imported_symbols = sym_table.imported_symbols(my_mod)
+    assert var3 in imported_symbols
+    assert var2 in imported_symbols
+    # Passing something that is not a ContainerSymbol is an error
+    with pytest.raises(TypeError) as err:
+        sym_table.imported_symbols(var2)
+    assert "expects a ContainerSymbol but got an object of type" in \
+        str(err.value)
+
+
+def test_remove():
+    '''Test that the remove method removes symbols from the symbol
+    table, but raises appropiate errors when provided with wrong
+    parameters. '''
+    sym_table = SymbolTable()
+
+    # Declare a symbol
+    my_mod = ContainerSymbol("my_mod")
+    sym_table.add(my_mod)
+    sym_table.add(DataSymbol("var1", DataType.REAL, shape=[5, 1],
+                             interface=GlobalInterface(my_mod)))
+    var1 = sym_table.lookup("var1")
+    assert var1
+    assert sym_table.imported_symbols(my_mod) == [var1]
+    # We should not be able to remove a Container if it is referenced
+    # by an existing Symbol
+    with pytest.raises(ValueError) as err:
+        sym_table.remove(my_mod)
+    assert ("Cannot remove ContainerSymbol 'my_mod' because symbols "
+            "['var1'] are imported from it" in str(err.value))
+    # Change the interface on var1
+    var1.interface = LocalInterface()
+    # We should now be able to remove the ContainerSymbol
+    sym_table.remove(my_mod)
+    with pytest.raises(KeyError) as err:
+        sym_table.lookup("my_mod")
+    assert "Could not find 'my_mod'" in str(err.value)
+    # Attempting to remove it a second time is an error
+    with pytest.raises(KeyError) as err:
+        sym_table.remove(my_mod)
+    assert ("Cannot remove Symbol 'my_mod' from symbol table because it does "
+            "not" in str(err.value))
+    # Attempt to supply something that is not a Symbol
```

Comment:
You could add a test to try to remove a symbol that is in the symbol table but is not a container symbol.

Reply:
Good idea. Done.